ARM: dts: qcom: msm8909-acer-t01: Add initial device tree

wonderfulShrineMaidenOfParadise opened the pull request:
Acer Liquid Z330 is a phone using the MSM8909 SoC released in 2015.

Add a device tree for with initial support for:

- GPIO keys
- pm8909-vibrator
- SDHCI (internal and external storage)
- USB Device Mode
- UART
- WCNSS (WiFi/BT)
- Regulators
- Touchscreen
- PWM panel backlight
- Flash LED
- Accelerometer
- Notification LED
- LBC and BMS

wonderfulShrineMaidenOfParadise committed:
ARM: dts: qcom: msm8909-acer-t01: Add initial device tree

Acer Liquid Z330 is a phone using the MSM8909 SoC released in 2015.

Add a device tree for with initial support for:

- GPIO keys
- pm8909-vibrator
- SDHCI (internal and external storage)
- USB Device Mode
- UART
- WCNSS (WiFi/BT)
- Regulators
- Touchscreen
- PWM panel backlight
- Flash LED
- Accelerometer
- Notification LED

Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
ARM: dts: qcom: msm8909-acer-t01: Add LBC and BMS

Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise commented on arch/arm/boot/dts/qcom-msm8909-acer-t01.dts:
```diff
@@ -0,0 +1,175 @@
+	l11 {
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <2950000>;
+		regulator-allow-set-load;
+		regulator-system-load = <200000>;
```
```suggestion
		regulator-system-load = <200000>;
		regulator-allow-set-load;
```

Not sure about this one.

wonderfulShrineMaidenOfParadise commented on drivers/leds/leds-aw2013.c:
```diff
@@ -348,16 +350,20 @@ static int aw2013_probe(struct i2c_client *client)
-	ret = PTR_ERR_OR_ZERO(chip->vcc_regulator);
-	if (ret) {
+	chip->regulators[0].supply = "vcc";
+	chip->regulators[1].supply = "vdd";
```
Todo: `iovcc` or `vccio`

wonderfulShrineMaidenOfParadise replied:
`vio`
Link: https://lore.kernel.org/lkml/20230815-aw2013-vio-v3-0-2505296b0856@gerhold.net/

stephan-gh commented on arch/arm/boot/dts/qcom-msm8909-acer-t01.dts:
```diff
@@ -36,6 +37,48 @@
+		interrupt-parent = <&tlmm>;
+		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+
+		reset-gpios = <&tlmm 12 GPIO_ACTIVE_LOW>;
```
Does that really work? Generally only one driver can request a GPIO. We might need to select this in lk2nd similar to idol3.

wonderfulShrineMaidenOfParadise replied:
Not really, since mine is not ft5306. I might resort the commit if it requires lk2nd selection.
However I cannot test it now before getting more reliable hardware.

stephan-gh replied:
Then please only add the touchscreen you have (and can test) for now.

stephan-gh commented on arch/arm/boot/dts/qcom/qcom-msm8909-acer-t01.dts:
```diff
@@ -141,6 +160,21 @@
+
+&pm8909_charger {
+	qcom,vdd-safe = <4150000>;
+	qcom,ibat-safe = <270000>;
```
Where does this come from downstream?

wonderfulShrineMaidenOfParadise replied:
They are copied from nokia-common.
https://github.com/msm8916-mainline/linux/commit/3b355bcdc20bf1f65b8d9199b17491b08ce3306a#diff-c50071839575c94d0d7d58c4d15ca086a27d0135ff82516c3acca15bbf48c013R182-R183

```suggestion
	qcom,vdd-safe = <4200000>;
	qcom,ibat-safe = <1440000>;
```

Will this be better?
https://github.com/msm8916-mainline/linux-downstream/blob/acer-z330/arch/arm/boot/dts/qcom/msm-pm8909.dtsi#L205-L207

TravMurav replied:
Hm, those values are often overriden a couple of times within the downstream rats-nest of dt...

Could you dump the downstream dtb blob and check the decompiled one? I'd prefer we keep those same or slightly lower than the downstream: i.e. the 1.44A charging limit sounds like an upper bound and the devices mostly have less than that set, and the battery charging voltage should for sure be double checked...

stephan-gh committed:
UPSTREAM: dt-bindings: leds: aw2013: Document interrupt

AW2013 has an optional interrupt pin "INTN" which is used to report
completion of started operations (e.g. power up or LED breath effects).
The driver does not use it (yet) but it should be described for
completeness inside the DT schema.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Reviewed-by: Nikita Travkin <nikita@trvn.ru>
Link: https://lore.kernel.org/r/20230815-aw2013-vio-v3-1-2505296b0856@gerhold.net
Signed-off-by: Lee Jones <lee@kernel.org>

stephan-gh committed:
UPSTREAM: dt-bindings: leds: Document pull-up supply for interrupt and I2C

Since the interrupt and I2C lines of AW2013 operate in open drain low
active mode a pull-up supply is needed for correct operation.
Unfortunately there is no ideal place to describe it in the DT: The
pull-up needed for the I2C lines could be described on the I2C bus.
However, the pull-up needed for the interrupt line belongs neither
directly to the interrupt controller nor to AW2013. Since the AW2013
driver will be typically in control of the power management and
interrupt masking it makes more sense to describe it inside the AW2013
device tree node.

Add it to the AW2013 DT schema together with a comment that makes it
clear what exactly it represents.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
Acked-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Reviewed-by: Nikita Travkin <nikita@trvn.ru>
Link: https://lore.kernel.org/r/20230815-aw2013-vio-v3-2-2505296b0856@gerhold.net
Signed-off-by: Lee Jones <lee@kernel.org>

bolilingmeng89006 committed:
UPSTREAM: leds: aw2013: Enable pull-up supply for interrupt and I2C

Request and enable the "vio" regulator that represents the power supply
that is needed for the pull-up resistors of the interrupt and I2C lines
of AW2013. While this regulator is not wired directly to the AW2013
chip it is best managed as part of the AW2013 driver since it decides
when AW2013 is powered on and when the interrupt is enabled or
disabled.

This regulator should always be enabled in conjunction with the main
VCC power supply, so use the bulk regulator functions to enable both
at the same time.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>
[rewrite commit message based on discussion]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
Reviewed-by: Nikita Travkin <nikita@trvn.ru>
Link: https://lore.kernel.org/r/20230815-aw2013-vio-v3-3-2505296b0856@gerhold.net
Signed-off-by: Lee Jones <lee@kernel.org>

stephan-gh commented:
Can you squash the commits like expected upstream nowadays? But keep the BMS/LBC commit separate for now since the driver isn't upstream yet.

stephan-gh commented:
Integrated this into `msm8916/6.5-rc7`. Thanks!
